web/src/main/java/org/wso2/testgrid/web/api/APIUtil.java

```diff
@@ -199,12 +207,14 @@ static TestScenario getTestScenarioBean(org.wso2.testgrid.common.TestScenario te
      */
     static TestCase getTestCaseBean(org.wso2.testgrid.common.TestCase testCase) {
         TestCase testCase1 = new TestCase();
-        testCase1.setId(testCase.getId());
-        testCase1.setName(testCase.getName());
-        testCase1.setSuccess(testCase.isSuccess());
-        testCase1.setModifiedTimestamp(testCase.getModifiedTimestamp());
-        testCase1.setCreatedTimestamp(testCase.getCreatedTimestamp());
-        testCase1.setErrorMsg(testCase.getFailureMessage());
+        if (testCase != null) {
+            testCase1.setId(testCase.getId());
```

Comment:
Shall we change the variable names `testCase` and `testCase1` to something more meaningful since it is hard to distinguish which one is which at the moment